website/docs/about_code_generation.mdx

```diff
@@ -55,13 +79,132 @@ This includes but is not limited to:
 - some Riverpod features will be available only with code generation.
 
 At the same time, many applications already use code generation with packages such
-as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).  
+as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).
 In that case, your project probably is already setup for code generation, and
 using for Riverpod should be simple.
 
+## The Syntax
+
+### Defining a provider:
+When defining a provider using code generation, there are two key points to consider:
+1. <span style={RED_STYLE}>Sync/Async</span> initialization: Choose the appropriate return type based on whether the state is initialized
```

Comment:
That description is a bit redundant. 

I think we should mention:
- all Dart async primitives are supported. So Future, FutureOr, and Stream
- Marking a function as async leads to the provider automatically handling errors/loading states and exposing an AsyncValue
- Async providers expose a `.future` modifier to allow dependent to `await` the operation